### PR TITLE
tests: change how pytest adds markers

### PR DESCRIPTION
**Pull request by mariuswinger**
## Why is this pull request needed?
The old setup didn't work particularly well/was buggy.

## What does this pull request change?
Markers are automatically added based on the parent test directory.
`pytest --unit` and `pytest` runs unit tests, while `pytest --integration` runs integration tests. Finally, `pytest --unit --integration` runs both. Furthermore, one can use markers to select the appropriate tests even quicker: e.g. `pytest -m integration`, which skips collecting all tests and just collects integration tests.

Also, pre-commit was failing on main, so I had to make some additional changes in order to make it run successfully. The state of this repo right now ...

## Issues related to this change:
None.

### CODE REVIEW

File: api/src/tests/conftest.py [hunk 1]
```diff
@@ -76,10 +76,50 @@ async def request_body():
...
+            if "unit" in item.keywords:
+                pytest.skip("unit tests are skipped when explicitly running integration tests")
+        case (True, False):
+            # skip integration tests
+            if "integration" in item.keywords:
+                pytest.skip("integration tests are skipped when explicitly running unit tests")
+        case (False, False):
+            # skip integration tests
+            if "integration" in item.keywords and "integration" not in marker_expr:
+                pytest.skip("need --integration option to run")
```
Comment (sebastianvitterso) [inline api/src/tests/conftest.py:125]: So if I don't have `--unit` or `--integration`, it will only run unit-tests? Why aren't all tests run?

**Review by sebastianvitterso — APPROVED** [review]
I can dig it

File: api/src/common/exceptions.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def __init__(
         self.extra = extra
         self.severity = severity
 
-    def dict(self) -> dict[str, int | str | dict[str, Any] | None]:
+    def model_dump(self) -> dict[str, int | str | dict[str, Any] | None]:
```
Comment (soofstad) [inline api/src/common/exceptions.py:47]: The pydantic warning you see when running pytest is not refering to this one.
By adding this to pyproject toml pytest:
```
filterwarnings = [
    "error"
]
```
Warning will raise errors, and you can see where they come from.

So, IMO, no need to rename this function. It's nice to differentiate it from a pydantic model

File: api/src/common/logger_level.py
Comment (soofstad) [inline api/src/common/logger_level.py:1]: Nice, but no need to redefine these.
`from logging import getLevelNamesMapping`

Comment (mariuswinger) [reply api/src/common/logger_level.py:1]: Only works in >Python 3.11, and we allow 3.10 in pyproject. Furthermore, it is a dict, which would then be more untyped with respect to the pydantic model where it is used.

**Pull request merged**